mod/item.php

```diff
@@ -636,15 +529,15 @@ function item_post(App $a) {
 
 	if (preg_match_all('/(\[attachment\]([0-9]+)\[\/attachment\])/',$body,$match)) {
 		foreach ($match[2] as $mtch) {
-			$r = q("SELECT `id`,`filename`,`filesize`,`filetype` FROM `attach` WHERE `uid` = %d AND `id` = %d LIMIT 1",
-				intval($profile_uid),
-				intval($mtch)
-			);
-			if (DBM::is_result($r)) {
+			$fields = ['id', 'filename', 'filesize', 'filetype'];
+			$attachment = dba::selectFirst('attach', $fields, ['id' => $mtch]);
```

Comment:
Why did you drop the `uid` condition?

Reply:
Same like before. The primary field is "id".

Reply:
Same like before, it made sure the attachment is owned by `$profile_uid`, can you still ensure it is? 😛

Reply:
I had the same idea, so I changed it.

Reply:
I've seen you added a check for the other attach condition, are you going to change this one as well?

Reply:
You could always want to link an attachment that has already been attached before (and it could be public)

Reply:
Ok, I'm just making sure we aren't losing a critical privacy check in the process. Are you done with your changes now?

Reply:
Yeah, I am. But currently my system blanks - but also on the "develop" branch, you it isn't a problem of my changes.

Reply:
Problem solved. It was in one of the addons.